src/main/python/rlbot/agents/standalone/executable_with_socket_standalone_bot.py

```diff
@@ -0,0 +1,72 @@
+import multiprocessing
+import os
+import socket
+import subprocess
+import sys
+from typing import Type
+
+import psutil
+from rlbot.agents.base_agent import BOT_CONFIG_AGENT_HEADER
+from rlbot.agents.executable_with_socket_agent import ExecutableWithSocketAgent
+from rlbot.agents.standalone.standalone_bot_config import StandaloneBotConfig
+from rlbot.parsing.bot_config_bundle import get_bot_config_bundle
+from rlbot.utils.process_configuration import configure_process, get_team_cpus
+
+
+class ExecutableWithSocketStandaloneBot(ExecutableWithSocketAgent):
+
+    def __init__(self, name, team, index):
+        super().__init__(name, team, index)
+
+    def find_process_using_port(self, port: int):
+        for proc in psutil.process_iter():
+            for conn in proc.connections():
+                if conn.laddr.port == port:
+                    return proc
+        return None
+
+    def is_port_open(self, port: int):
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            s.connect(("127.0.0.1", port))
+            s.close()
+            return True
+        except ConnectionRefusedError:
+            s.close()
+            return False
+
+    def send_add_command(self):
+        message = self.build_add_command()
+        self.logger.info(f"About to send add message for {self.name}")
+        return self.send_command(message)
+
+    def ensure_active_executable(self):
+        self.logger.info("Ensuring active executable...")
+        #
+        # if self.is_port_open(self.get_port()):
+        #     self.logger.info("Found port already available")
+        # elif self.find_process_using_port(self.get_port()):
+        #     self.logger.info("Found process already running")
+        # else:
+        launch = [self.executable_path, str(self.get_port())]
+        proc = subprocess.Popen(launch, cwd=os.path.dirname(self.executable_path))
+        process = psutil.Process(pid=proc.pid)
+        configure_process(process, get_team_cpus(self.team))
+        # TODO: handle the case where the process spans both teams; should be pinned differently
+        self.logger.info("Launched executable instance")
```

Comment:
There's a TODO here. Is this something that needs fixing before release?

Reply:
Good catch! I wrote this part of the code half a year ago and forgot, I'll deal with it